src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -569,11 +575,26 @@ public function next_token() {
 	 * @return bool Whether the current tag is a tag closer.
 	 */
 	public function is_tag_closer() {
-		return isset( $this->current_element )
-			? ( WP_HTML_Stack_Event::POP === $this->current_element->operation )
+		return $this->is_virtual()
+			? ( WP_HTML_Stack_Event::POP === $this->current_element->operation && '#tag' === $this->get_token_type() )
 			: parent::is_tag_closer();
 	}
 
+	/**
+	 * Indicates if the currently-matched token is virtual, created by a stack operation
+	 * while processing HTML, rather than a token found in the HTML text itself.
+	 *
+	 * @since 6.6
+	 *
+	 * @return bool Whether the current token is virtual.
+	 */
+	private function is_virtual() {
```

Comment:
Interesting. Since it's private there is no new testing code coverage necessary. 

However, we probably should add some tests that cover the cases when overridden methods are called on virtual nodes to ensure they work as intended.